[Build] Update SWIFT_TARGET_LIBRARY_NAME for swiftRuntime in new build.

al45tair opened the pull request:
We need to change `SWIFT_TARGET_LIBRARY_NAME` for the `swiftRuntime` target in the new build because of changes made to the old build system.

al45tair commented:
@swift-ci Please smoke test

al45tair commented:
See #78516, which changed the name so that we can use `Runtime` as a Swift module name.

etcwilde commented on Runtimes/Core/runtime/CMakeLists.txt:
```diff
@@ -83,10 +83,10 @@ endif()
   PRIVATE
     -DSWIFT_RUNTIME
-    -DSWIFT_TARGET_LIBRARY_NAME=swiftRuntime
+    -DSWIFT_TARGET_LIBRARY_NAME=swiftRuntimeCore
```
I believe this is the only part that actually needs changing here so that the compatibility header picks things up correctly. This is an object library that only contains C/C++, so there shouldn't be any conflict with parts from the old build. Given that the `Runtime` module depends on the overlay, it will have to be a separate "supplemental" library anyway.

al45tair committed:
[Build] Update SWIFT_TARGET_LIBRARY_NAME for swiftRuntime in new build.

We need to change `SWIFT_TARGET_LIBRARY_NAME` for the `swiftRuntime`
target in the new build because of changes made to the old build system.

al45tair commented:
@swift-ci Please smoke test

compnerd commented on Runtimes/Core/runtime/CMakeLists.txt:
```diff
@@ -86,7 +86,7 @@ endif()
   PRIVATE
     -DSWIFT_RUNTIME
-    -DSWIFT_TARGET_LIBRARY_NAME=swiftRuntime
+    -DSWIFT_TARGET_LIBRARY_NAME=swiftRuntimeCore
```
This is incorrect, this should be `swiftCore`. The way that this macro is used is with the CPP macro programming in `Visibility.h`, and the only valid options are `swiftCore`, `swift_Concurrency`, `swiftDistributed, `swift_Differentiation`. The value is used to select if the symbol is in the current module's ABI or not.
